### PR TITLE
Add ASP.NET example for .NET SDK

### PR DESCRIPTION
**Pull request by edif2008**
This PR contains the example for using the SecretHub XGO library in an Azure App.

### CODE REVIEW

File: application/azure/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from mcr.microsoft.com/dotnet/core/sdk:3.1
```
Comment (florisvdg) [inline application/azure/Dockerfile:1]:
```suggestion
FROM mcr.microsoft.com/dotnet/core/sdk:3.1
```

File: application/azure/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+RUN mkdir azure
+WORKDIR /azure
+RUN dotnet new web
+COPY *.csproj *.cs *.nupkg ./
+RUN dotnet add package SecretHub -s $PWD
+
+# Install SecretHub
+RUN echo "deb [trusted=yes] https://apt.secrethub.io stable main" > /etc/apt/sources.list.d/secrethub.sources.list && apt-get update
+RUN apt-get install -y secrethub-cli
```
Comment (florisvdg) [inline application/azure/Dockerfile:11]: Considering Docker's caching and layers system, it's better to first install any dependencies and then `COPY` source code:

```suggestion
# Install SecretHub
RUN echo "deb [trusted=yes] https://apt.secrethub.io stable main" > /etc/apt/sources.list.d/secrethub.sources.list && apt-get update
RUN apt-get install -y secrethub-cli

RUN mkdir azure
WORKDIR /azure
RUN dotnet new web
COPY *.csproj *.cs *.nupkg ./
RUN dotnet add package SecretHub -s $PWD
```

**Review by florisvdg — COMMENTED** [review]
Some quick comments on the Dockerfile

File: application/azure/Dockerfile (hunk 1 above)
Comment (edif2008) [reply application/azure/Dockerfile:1]: Fixed

File: application/azure/Dockerfile (hunk 2 above)
Comment (edif2008) [reply application/azure/Dockerfile:11]: Fixed

**Comment by florisvdg** [comment]
This is an example just for Azure App Service, right? So the path should be `azure/app-service/Program.cs` (also without `application`).

**Review by florisvdg — CHANGES_REQUESTED** [review]
And also, for cloud examples, it should contain deployment instructions and/or Terraform files if applicable.

**Comment by jpcoenen** [comment]
This is waiting for the public beta of the .NET SDK. Until then, I am marking this as a drat.

**Comment by Marton6** [comment]
Closing as the example from this PR has been improved moved to a different branch. It will be merged in this PR:
https://github.com/secrethub/examples/pull/52

**Pull request closed without merging**